qiskit/ignis/experiments/calibration/instruction_data/gate_and_schedules_definition.py

```diff
@@ -55,243 +78,196 @@ def from_backend(cls, backend: BaseBackend) -> 'InstructionsDefinition':
         Returns:
             New InstructionSet instance.
         """
-        parameter_library = dict()
-        channel_qubit_map = dict()
+        # TODO implement this
+        # we need a logic to convert Schedule into database code.
+        # Schedule doesn't have context. Thus we need to infer the context.
+
+        raise NotImplementedError
+
+        # parameter_library = dict()
+        # channel_qubit_map = dict()
+        #
+        # for chname, ch_properties in backend.configuration().channels.items():
+        #     channel_qubit_map[chname] = tuple(ch_properties['operates']['qubits'])
+        #
+        # pulse_table, sched_template = parse_backend_instmap(
+        #     channel_qubit_map=channel_qubit_map,
+        #     instmap=backend.defaults(refresh=True).instruction_schedule_map,
+        #     parameter_library=parameter_library)
+        #
+        # return InstructionsDefinition(
+        #     backend_name=backend.name(),
+        #     n_qubits=backend.configuration().n_qubits,
+        #     channel_qubit_map=channel_qubit_map,
+        #     pulse_table=pulse_table)
 
-        for chname, ch_properties in backend.configuration().channels.items():
-            channel_qubit_map[chname] = tuple(ch_properties['operates']['qubits'])
+    @property
+    def parametric_shapes(self) -> Enum:
+        """Return the definition of parametric pulses."""
+        return self._parametric_shapes
 
-        pulse_table, sched_template = parse_backend_instmap(
-            channel_qubit_map=channel_qubit_map,
-            instmap=backend.defaults(refresh=True).instruction_schedule_map,
-            parameter_library=parameter_library)
+    @property
+    def pulse_channels(self) -> Enum:
+        """Return the definition of pulse channels."""
+        return self._pulse_channels
 
-        return InstructionsDefinition(
-            backend_name=backend.name(),
-            n_qubits=backend.configuration().n_qubits,
-            channel_qubit_map=channel_qubit_map,
-            pulse_table=pulse_table)
+    @property
+    def series(self) -> str:
+        """Return the current data series to construct gate schedule."""
+        return self._series
```

Comment:
Is `series` and `data series` really the best wording here? This sounds a lot like `series` in the `Analysis` part of calmod.

Reply:
Good point. I agree with you. Do you have better wording for this?

Reply:
No sure, what is the purpose of `series`? Is `context` or `scope` better?

Reply:
Defining another set of parameters. For example, if I invent new calibration technique, I want to run calibration with standard ping-pong and my technique while keeping the same pulse structure. In this case I may name conventional one as `standard` and new as `my_cal`. I want store two sets of parameters for all gate schedules and generate RB experiment from these sets. By using `series` we can easily switch the parameter set to generate gates.